### PR TITLE
Cleanup circleci binary generation

### PR DESCRIPTION
**Pull request by atalman**
Cleanup circleci binary generation

### CODE REVIEW

**Review by pmeier — APPROVED** [review]
Thanks! Only CMake left :rocket:

File: .circleci/regenerate.py [hunk 1]
```diff
@@ -21,201 +21,6 @@
...
-        "name": f"{base_workflow_name}_{smoke_suffix}",
-        "requires": [required_build_name],
-        "python_version": python_version,
-    }
-
-    if filter_branch:
-        d["filters"] = gen_filter_branch_tree(filter_branch)
-
-    return {f"smoke_test_{os_type}_{pydistro}": d}
-
```
Comment (pmeier) [inline .circleci/regenerate.py:24]: See https://github.com/pytorch/vision/actions/runs/4984519684/jobs/8922956818?pr=7591

```suggestion
```

File: .circleci/config.yml.in [hunk 2]
```diff
@@ -124,55 +124,7 @@ jobs:
...
-      - store_artifacts:
-          path: dist
-      - persist_to_workspace:
-          root: dist
-          paths:
-            - "*"
-      - store_test_results:
-          path: build_results/
 
   binary_macos_wheel:
```
Comment (osalpekar) [inline .circleci/config.yml.in:129]: These jobs are not being used either. Should we just clean them up as well?

**Review by osalpekar — APPROVED** [review]

File: .circleci/config.yml.in (hunk 2 above)
Comment (atalman) [reply .circleci/config.yml.in]: Done! Cleaner these up!

**Pull request merged**